plugins/optimization/src/main/java/org/qbicc/plugin/opt/InliningBasicBlockBuilder.java

```diff
@@ -84,14 +82,12 @@ public Value call(ValueHandle target, List<Value> arguments) {
         ExecutableElement toInline = getInlinedElement(target);
         if (toInline != null) {
             BlockLabel resumeLabel = new BlockLabel();
-            PhiValue returnVal = phi(toInline.getType().getReturnType(), resumeLabel);
-            BasicBlock inlined = doInline(target, toInline, arguments, null, val -> {
-                BasicBlock basicBlock = goto_(resumeLabel, Map.of());
-                returnVal.setValueForBlock(ctxt, toInline, basicBlock, val);
-                return basicBlock;
-            }, () -> begin(resumeLabel));
+            BasicBlock inlined = doInline(target, toInline, arguments, null, val -> goto_(resumeLabel, Slot.result(), val), () -> {
+                begin(resumeLabel);
+                addParam(resumeLabel, Slot.result(), toInline.getType().getReturnType());
+            }, Map.of());
             if (inlined != null) {
-                return returnVal;
+                return null; // todo: Slot.result()
```

Comment:
is returning `null` a safe enough "this isn't done yet bail our" or should we be throwing an exception?

Reply:
Inlining is basically completely broken at the moment so it doesn't make a difference right now. Block parameters should allow a cleaner implementation but I didn't want to clutter up this change with yet more craziness.